### PR TITLE
fix checkout callback during add/commit

### PR DESCRIPTION
**Pull request by skshetry**
Fix https://github.com/iterative/dvc/issues/6226.

Current dependencies on/for this PR:

- https://github.com/iterative/dvc-objects/pull/208 
    - https://github.com/iterative/dvc-data/pull/355
        - https://github.com/iterative/dvc/pull/9445 👈🏼 

## TODO:
- [x] Release `dvc-objects`
- [x] Release `dvc-data`

### CODE REVIEW

File: dvc/output.py [hunk 1]
```diff
@@ -890,9 +894,19 @@ def checkout(
         checkpoint_reset: bool = False,
         **kwargs,
     ) -> Optional[Tuple[bool, Optional[bool]]]:
+        # callback passed act as a aggregate callback.
+        # do not let checkout to call set_size and change progressbar.
+        class CallbackProxy(Callback):
```
Comment (skshetry) [inline dvc/output.py:899]: I don't like this that it passes an aggregated callback, but this seemed easier to do for now.

**Comment by skshetry** [comment]
@efiop, I am getting:
```console
dvc/repo/worktree.py: note: In function "push_worktree":
dvc/repo/worktree.py:176:27: error: Unsupported operand types for + ("int" and
"Dict[str, List[Change]]")  [operator]
                    pushed += checkout(
                              ^
Found 1 error in 1 file (checked 283 source files
```

What is the equivalent to get total no of push now?

**Comment by efiop** [comment]
@skshetry Just sum up all the changes (it returns a dict of lists, just sum up length of all lists) for now. That worktree code will go away soon in https://github.com/iterative/dvc/pull/9424 anyway.

**Comment by skshetry** [comment]
There's unchanged too, right?

**Comment by efiop** [comment]
@skshetry No, this is index diff not hashfile diff, it has `with_unchanged` as an option that is not always turned on (at least right now).

**Comment by skshetry** [comment]
Thanks, updated.

**Pull request merged**